### DISCUSSION TITLE
Why kong access non configured regex route

### DISCUSSION THREAD

**Discussion by khotaditya** [Kong/kong#7838] [Help] [answered]
Hello,

I'm using Kong v2.1.2

On my upstream server I have APIs
1. GET /v1/country
2. GET /v1/country/{country_code}/brands
3. GET /v1/country/{country_code}/brands/{brand_code}

I have configured path 1 and 2 on kong routes with host header "example.com"
I can able to access 1 and 2 APIs using Kong.
But the interesting thing is, I can access 3rd API too with same host header even though it is not configured in Kong.

So the question is how Kong can access those APIs which are not configured and how can I disallow the requests which are not configured on kong but present in upstream server.

Please help me to understand this.

Thank you!

**Comment by kikito** [comment, marked as answer]
Hello @khotaditya ,

Routes allow using regular expressions on their paths. With this in mind, you could use regexes that end in `$` so that they match the "end of the line", for example: `^\/v1\/country$` and `^\/v1\/country\/\w+\/brands$`
